### PR TITLE
Fixed the layer scaling issue on macOS.

### PR DESCRIPTION
**Pull request by golddranks**
Fixed the layer scaling issue on macOS, as `metal-rs` published the needed API.

Also changed the triangle example to use the surface extents instead of window dimensions, as that seems to be a more robust source of correct information. (At least one user reported that on their X.org-based system the surface was created according to the outer dimensions of the window, not inner, which is contrary to macOS. I don't know about Windows.)

The triangle example now runs straight out of the box on my macOS-based system, if the MoltenVK dynamic library is installed. Also, according to the MoltenVK support forums, the next version is going to fix the excessive logging.

Fixes https://github.com/vulkano-rs/vulkano/issues/738

### CODE REVIEW

**Comment by tomaka** [comment]
Could you add a line in `CHANGELOG.md` about the fix in vulkano-win?

The fix to the example should also be propagated to the vulkano-examples repository eventually (not necessarily by you).

**Comment by golddranks** [comment]
I updated the changelog. I also propagated the fix to the other in-repo examples, but haven't looked at the vulkano-examples repo at all.

File: examples/src/bin/.vscode/tasks.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
+{
```
Comment (tomaka) [inline examples/src/bin/.vscode/tasks.json]: Please remove

Comment (golddranks) [reply examples/src/bin/.vscode/tasks.json]: Ugh sorry. I hate when VS Code does that.

**Comment by golddranks** [comment]
I `git commit --amend` and `git push --force`d that to prevent littering the history with a unrelated stuff like that.

**Pull request merged**